src/skadi/sample.cc

```diff
@@ -456,9 +443,8 @@ template <class coord_t> double sample::get(const coord_t& coord) {
 
   // get the proper source of the data
   const auto& tile = cache_->source(index);
-  if (!tile) {
-    return NO_DATA_VALUE;
-  }
+  if (!tile)
+    return get_from_remote(coord);
```

Comment:
I believe it's better to implement it like this
```cpp
const auto& tile = cache_->source(index);
if (!tile) {
  // fetch tile
  fetch_from_remote(tile);
  // try to load it from the filesystem one more time
  tile = cache_->source(index);
}
```
In this case you don't need to have any coordinate-related logic in the fetch_from_remote/get_from_remote: it will do pure fetching

Reply:
yeah, it seems it will look better